src/npcmove.cpp

```diff
@@ -2375,6 +2398,26 @@ bool npc::do_pulp()
     return moves != old_moves;
 }
 
+bool npc::do_player_activity()
+{
+    int old_moves = moves;
+    while( moves > 0 && activity ) {
+        activity.do_turn( *this );
+    }
+    /* if the activity is finished, grab any backlog or change the mission */
+    if( !has_destination() && !activity ) {
```

Comment:
Should this be `has_destination_activity()` instead of `has_destination()`?

Reply:
You'd think that, wouldn't you?  I sure did.

has_destination_activity() only returns true when you have a destination_activity and you've reached your destination for automove.  It's used to detect that you can resume an activity that was interrupted for movement.

This is a check that an activity has completed entirely, so no activity and no autoroute movement is the right check.